server/events.go

```diff
@@ -327,6 +327,15 @@ type ClientInfo struct {
 	Nonce      string        `json:"nonce,omitempty"`
 }
 
+// forAssignmentSnap returns the minimum amount of ClientInfo we need for assignment snapshots.
+func (ci *ClientInfo) forAssignmentSnap() *ClientInfo {
+	return &ClientInfo{
+		Account: ci.Account,
```

Comment:
So because the returned value is another `ClientInfo`, the `serviceAccount()` function will continue to work just the same, which is that it returns `Service` if set and `Account` if not.